tests/v2/validation/charts/backup_restore/backup_restore.go

```diff
@@ -0,0 +1,331 @@
+	userList := []*management.User{}
+	projList := []*management.Project{}
+	roleList := []*management.RoleTemplate{}
+
+	for i := 0; i < resourceCount; i++ {
+		u, err := users.CreateUserWithRole(client, users.UserConfig(), "user")
+		if err != nil {
+			return userList, projList, roleList, err
+		}
+		userList = append(userList, u)
+
+		p, _, err := projects.CreateProjectAndNamespace(client, clusterID)
+		if err != nil {
+			return userList, projList, roleList, err
+		}
+		projList = append(projList, p)
+
+		rt, err := client.Management.RoleTemplate.Create(
+			&management.RoleTemplate{
+				Context: context,
+				Name:    roleName,
+				Rules:   rules,
+			})
+		if err != nil {
+			return userList, projList, roleList, err
+		}
+		roleList = append(roleList, rt)
+	}
+
+	return userList, projList, roleList, nil
+}
+
+func createBackup(client *rancher.Client, bucket string) (*v1.SteveAPIObject, string, error) {
+	backup := setBackupObject()
+	backupTemplate := bv1.NewBackup("", backupName, backup)
+	completedBackup, err := client.Steve.SteveType(backupSteveType).Create(backupTemplate)
+	if err != nil {
+		return nil, "", err
+	}
+
+	charts.VerifyBackupCompleted(client, backupSteveType, completedBackup)
+
+	backupObj, err := client.Steve.SteveType(backupSteveType).ByID(backupName)
+	if err != nil {
+		return nil, "", err
+	}
+
+	backupK8Obj := &bv1.Backup{}
+	err = v1.ConvertToK8sType(backupObj.JSONResp, backupK8Obj)
+	if err != nil {
+		return nil, "", err
+	}
+
+	backupFileName := backupK8Obj.Status.Filename
+
+	client.Session.RegisterCleanupFunc(func() error {
+		err = deleteAWSObject(bucket, backupFileName)
+		if err != nil {
+			return err
+		}
+
+		_, err = checkAWSS3Object(bucket, backupFileName)
+		if err != nil {
+			s3Error := err.Error()
+			errorText := strings.Contains(s3Error, "404")
+			if !errorText {
+				return err
+			}
+		}
+
+		return nil
+	})
+
+	return completedBackup, backupFileName, err
+}
+
+func createDownstreamCluster(client *rancher.Client, clusterType string) (*management.Cluster, *v1.SteveAPIObject, *clusters.ClusterConfig, error) {
+	provisioningConfig := new(provisioninginput.Config)
+	config.LoadConfig(provisioninginput.ConfigurationFileKey, provisioningConfig)
+	nodeProviders := provisioningConfig.NodeProviders[0]
+	externalNodeProvider := provisioning.ExternalNodeProviderSetup(nodeProviders)
+	testClusterConfig := clusters.ConvertConfigToClusterConfig(provisioningConfig)
+
+	var clusterObject *management.Cluster
+	var steveObject *v1.SteveAPIObject
+	var err error
+
+	switch clusterType {
+	case "RKE1":
+		nodeAndRoles := []provisioninginput.NodePools{
+			provisioninginput.AllRolesNodePool,
+		}
+		testClusterConfig.NodePools = nodeAndRoles
+		testClusterConfig.KubernetesVersion = provisioningConfig.RKE1KubernetesVersions[0]
+		clusterObject, _, err = provisioning.CreateProvisioningRKE1CustomCluster(client, &externalNodeProvider, testClusterConfig)
```

Comment:
is there somewhere in the readme you say it has to be custom clusters?

Reply:
no, cluster type doesn't really matter as long as we have one RKE1 and one RKE2 cluster.

Reply:
since you are explicitly calling the customCluster create call though, I think it would make sense to either update the function to handle custom and provisioned cluster creation, or update the readme to explicitly instruct the user to only provide / expect custom clusters configs.